Task_DEV_4/Task_DEV_4/Triangle.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+
+namespace Task_DEV_4
+{
+    class Triangle : Figure
+    {
+        private const string ERROR_MESSAGE = "Side cannot be less equals than 0.";
+        private const string ERROR_MESSAGE2 = "The sum of the two sides cannot be less than the third side.";
+
+        public double BaseSide { get; set; }
+        public double FirstSide { get; set; }
+        public double SecondSide { get; set; }
+
+        public Triangle(string color, string name, double baseSide, double firstSide, double secondSide) : base(color, name)
+        {
+            BaseSide = baseSide;
+            FirstSide = firstSide;
+            SecondSide = secondSide;
+            CheckForExistance();
+        }
+
+        public override double CalculateArea()
+        {
+            double perimeter = BaseSide + FirstSide + SecondSide;
+            double semiPerimeter = perimeter / 2;
+            return Math.Sqrt(semiPerimeter * (semiPerimeter - BaseSide) *
+                (semiPerimeter - FirstSide) * (semiPerimeter - SecondSide));
+        }
+
+        public override double CalculatePerimeter()
+        {
+            return BaseSide + FirstSide + SecondSide;
+        }
+
+        public override void CheckForExistance()
```

Comment:
which came first, the chicken or the egg?:) (think about order of checks)

Reply:
Done.